fvm/systemcontracts/system_contracts.go

```diff
@@ -45,9 +45,10 @@ const (
 
 	// Unqualified names of service events (not including address prefix or contract name)
 
-	EventNameEpochSetup    = "EpochSetup"
-	EventNameEpochCommit   = "EpochCommit"
-	EventNameVersionBeacon = "VersionBeacon"
+	EventNameEpochSetup                  = "EpochSetup"
+	EventNameEpochCommit                 = "EpochCommit"
+	EventNameVersionBeacon               = "VersionBeacon"
+	EventNameProtocolStateVersionUpgrade = "ProtocolStateVersionUpgrade"
```

Comment:
I think the different names are fine for now. After all, we want to experiment with a new approach. If we have worked through the details, we can consolidate both mechanisms.

Reply:
> Protocol State can safely change the version at any time without delay; VersionBeacon has the "versionUpdateFreezePeriod", because of sealing lag.

I'm not sure I understand why that is? The `ProtocolStateVersionUpgrade` event could have been emitted on a fork that will get discarded. Is this particular difference between `ProtocolStateVersionUpgrade` and `EventNameVersionBeacon` documented somewhere?